xlators/cluster/afr/src/afr-inode-read.c

```diff
@@ -879,14 +879,14 @@ afr_update_local_dicts(call_frame_t *frame, dict_t *dict, dict_t *xdata)
 
     ret = -ENOMEM;
 
-    if ((dict != NULL) && (local->dict == NULL)) {
```

Comment:
I think it would be good to retain this check, because as per the condition at the calling function
`if ((op_ret >= 0) && ((lockinfo != NULL) || (xdata != NULL)))`
there is a possibility that either dict or xdata might be null. In that case we need not have to allocate mem for the one which is null.